satpy/tests/test_scene.py

```diff
@@ -222,6 +222,38 @@ def test_create_reader_instances_with_reader_kwargs(self):
             self.assertDictEqual(reader_kwargs, r.create_filehandlers.call_args[1]['fh_kwargs'])
             del scene
 
+    def test_create_multiple_reader_different_kwargs(self):
+        """Test passing different kwargs to different readers."""
+        from satpy.scene import Scene
+        from satpy.tests.utils import FakeReader
+        with mock.patch('satpy.readers.configs_for_reader') as src, \
+             mock.patch("satpy.readers.load_reader") as srl:
+            r1 = FakeReader('strona')
+            r1.select_files_from_pathnames = mock.MagicMock()
+            r1.select_files_from_pathnames.return_value = ["campello monti"]
+            r1.create_filehandlers = mock.MagicMock()
+            r2 = FakeReader('mastallone')
+            r2.select_files_from_pathnames = mock.MagicMock()
+            r2.select_files_from_pathnames.return_value = ["rimella"]
+            r2.create_filehandlers = mock.MagicMock()
+
+            def fake_lr(reader_configs, **reader_kwargs):
+                if reader_configs == ["/fake/strona.yaml"]:
+                    return r1
+                elif reader_configs == ["/fake/mastallone.yaml"]:
+                    return r2
+                else:
+                    raise RuntimeError("unit test is broken")
```

Comment:
Could you just not put an `else` here and if `None` is returned (the default when no return statement is encountered) then the rest of the code should fail.

Reply:
Yes, I could.  Why would that be better?

Reply:
It would fix the not covered line and would *maybe* be less awkward so the test doesn't have to test itself.

Reply:
Ok, I removed the `else:`-branch.